google/resource_logging_project_sink_test.go

```diff
@@ -0,0 +1,203 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"google.golang.org/api/logging/v2"
+	"strconv"
+	"testing"
+)
+
+func TestAccLoggingProjectSink_basic(t *testing.T) {
+	sinkName := "tf-test-sink-" + acctest.RandString(10)
+	bucketName := "tf-test-sink-bucket-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLoggingProjectSinkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccLoggingProjectSink_basic(sinkName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLoggingProjectSink(
+						"google_logging_project_sink.basic"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccLoggingProjectSink_uniqueWriter(t *testing.T) {
+	sinkName := "tf-test-sink-" + acctest.RandString(10)
+	bucketName := "tf-test-sink-bucket-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLoggingProjectSinkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccLoggingProjectSink_uniqueWriter(sinkName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLoggingProjectSink(
+						"google_logging_project_sink.unique_writer"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccLoggingProjectSink_updateUniqueWriter(t *testing.T) {
```

Comment:
Isn't this updating the bucket (/ the destination), not the unique writer?

Reply:
Yeah, it is. The test is poorly named. I changed it to '_updatePreservesUniqueWriter' (as that's explicitly tested)